R/pam-tidiers.R

```diff
@@ -0,0 +1,91 @@
+' Tidying methods for pam objects
+#' 
+#' These methods summarize the results of Partitioning Around Medoids (PAM) clustering into three
+#' tidy forms. \code{tidy} describes the medoid and size of each cluster,
+#' \code{augment} adds the cluster assignments to the original data, and
+#' \code{glance} summarizes the average silhouette with of the clustering.
+#'
+#' @param x pam object
+#' @param data Original data (required for \code{augment})
+#' @param col.names The names to call each dimension of the data in \code{tidy}.
+#' Defaults to \code{x1, x2...}
+#' @param ... extra arguments, not used
+#'
+#' @return All tidying methods return a \code{data.frame} without rownames.
+#' The structure depends on the method chosen.
+#'
+#' @seealso \code{\link{pam}}
+#'
+#' @examples
+#' 
+library(dplyr)
+library(ggplot2)
+library(cluster)
+
+set.seed(2014)
+centers <- data.frame(cluster=factor(1:3), size=c(100, 150, 50),
+                      x1=c(5, 0, -3), x2=c(-1, 1, -2))
+points <- centers %>% group_by(cluster) %>%
```

Comment:
Let's do an example that uses more modern tidyverse verbs instead of `do()` and `data.frame()`